src/pybind/mgr/cephadm/services/nfs.py

```diff
@@ -115,71 +118,37 @@ def get_set_cmd_dicts(out: str) -> List[dict]:
             get_set_cmd_dicts=get_set_cmd_dicts
         )
 
-
-class NFSGanesha(object):
-    def __init__(self,
-                 mgr: "CephadmOrchestrator",
-                 daemon_id: str,
-                 spec: NFSServiceSpec) -> None:
-        assert spec.service_id and daemon_id.startswith(spec.service_id)
-        mgr._check_pool_exists(spec.pool, spec.service_name())
-
-        self.mgr = mgr
-        self.daemon_id = daemon_id
-        self.spec = spec
-
-    def get_daemon_name(self) -> str:
-        return '%s.%s' % (self.spec.service_type, self.daemon_id)
-
-    def get_rados_user(self) -> str:
-        return '%s.%s' % (self.spec.service_type, self.daemon_id)
-
-    def get_keyring_entity(self) -> str:
-        return utils.name_to_config_section(self.get_rados_user())
-
-    def get_or_create_keyring(self, entity: Optional[str] = None) -> str:
-        if not entity:
-            entity = self.get_keyring_entity()
-
+    def get_or_create_keyring(self, entity: str) -> str:
         logger.info('Create keyring: %s' % entity)
-        ret, keyring, err = self.mgr.mon_command({
+
+        ret, keyring, err = self.mgr.check_mon_command({
             'prefix': 'auth get-or-create',
             'entity': entity,
         })
 
-        if ret != 0:
-            raise OrchestratorError(
-                'Unable to create keyring %s: %s %s'
-                % (entity, ret, err))
         return keyring
 
-    def update_keyring_caps(self, entity: Optional[str] = None) -> None:
-        if not entity:
-            entity = self.get_keyring_entity()
-
-        osd_caps = 'allow rw pool=%s' % (self.spec.pool)
-        if self.spec.namespace:
-            osd_caps = '%s namespace=%s' % (osd_caps, self.spec.namespace)
+    def update_keyring_caps(self, entity: str, spec: NFSServiceSpec) -> None:
+        osd_caps = 'allow rw pool=%s' % (spec.pool)
+        if spec.namespace:
+            osd_caps = '%s namespace=%s' % (osd_caps, spec.namespace)
 
         logger.info('Updating keyring caps: %s' % entity)
-        ret, out, err = self.mgr.mon_command({
+        ret, out, err = self.mgr.check_mon_command({
```

Comment:
No need to save the return values.
```suggestion
        self.mgr.check_mon_command({
```